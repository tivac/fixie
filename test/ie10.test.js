"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie10", () => {
    it("should transform :ie10(...) into an ie10 hack (single selector)", () =>
        simple(`
            :ie10(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie10(...) into an ie10 hack (multiple selectors)", () =>
        simple(`
            :ie10(.fooga .wooga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie10(&) {
                    color: blue;
                }
            }
        `)
    );
});
