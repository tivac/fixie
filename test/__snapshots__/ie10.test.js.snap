// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie10 should transform :ie10(...) into an ie10 hack (multiple selectors) 1`] = `
"_:-ms-lang(x),.fooga .wooga {
    color: red\\\\9;
}"
`;

exports[`:ie10 should transform :ie10(...) into an ie10 hack (single selector) 1`] = `
"_:-ms-lang(x),.fooga {
    color: red\\\\9;
}"
`;

exports[`:ie10 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    _:-ms-lang(x),.fooga {
        color: blue\\\\9;
    }"
`;
